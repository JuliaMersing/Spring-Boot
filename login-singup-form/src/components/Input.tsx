import React, { FormEvent } from 'react';

type InputProps = {
  value: string;
  onChange: (e: FormEvent) => any
  onBlur?: (e: FormEvent) => any
  type: string,
  id: string,
  placeholder: string,
  dataTestId: string,
  error?: string
  className: string
}

const Input = ({
  value,
  onChange,
  onBlur,
  type,
  id,
  placeholder,
  dataTestId,
  error,
  className,
}: InputProps) => (
  <div className="mb-6">
    <input
      onChange={onChange}
      onBlur={onBlur}
      value={value}
      className={className}
      type={type}
      id={id}
      placeholder={placeholder}
      {...{ 'data-testid': `${dataTestId}-input` }}
    />
    {error && (
      <div className="text-red-600">{error}</div>
    )}
  </div>
);

export default Input;
