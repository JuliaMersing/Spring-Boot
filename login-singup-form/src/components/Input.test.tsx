import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import Input from './Input';

describe('Input', () => {
  const setUp = () => {
    const mockHandleOnChange = jest.fn();
    const utils = render(<Input
      onChange={(e) => {
        mockHandleOnChange();
      }}
      value=""
      type="email"
      id="email"
      placeholder="Email address"
      dataTestId="email"
      className="input"
    />);
    return {
      ...utils,
      mockHandleOnChange,
    };
  };

  test('should match snapshot', () => {
    setUp();
    expect(screen.getByRole('textbox', { name: '' }))
      .toMatchSnapshot();
  });
  test('pass valid email to test email input field', async () => {
    const {
      mockHandleOnChange,
    } = setUp();
    const email = 'test@mail.com';

    const input = screen.getByRole('textbox', { name: '' });
    await userEvent.type(input, email);

    expect(mockHandleOnChange)
      .toHaveBeenCalled();
  });
});
