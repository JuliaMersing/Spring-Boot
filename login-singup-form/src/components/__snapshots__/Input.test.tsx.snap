// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<input
  class="input"
  data-testid="email-input"
  id="email"
  placeholder="Email address"
  type="email"
  value=""
/>
`;
